python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,202 @@
+        if response['ResponseMetadata']['HTTPStatusCode'] != 200:
+            raise ValueError(f"Got unexpected status code {response['HttpStatusCode']}")
+
+    def _glue_to_iceberg(self, glue_table, io: FileIO) -> Table:
+        properties: Dict[str, str] = glue_table["Parameters"]
+
+        if "table_type" not in properties:
+            raise NoSuchTableError(
+                f"Property table_type missing, could not determine type: {glue_table['DatabaseName']}.{glue_table['Name']}")
+        glue_table_type = properties.get("table_type")
+        if glue_table_type.lower() != ICEBERG:
+            raise NoSuchTableError(
+                f"Property table_type is {glue_table_type}, expected {ICEBERG}: {glue_table['DatabaseName']}.{glue_table['Name']}")
+        if prop_meta_location := properties.get(METADATA_LOCATION):
+            metadata_location = prop_meta_location
+        else:
+            raise NoSuchTableError(f"Table property {METADATA_LOCATION} is missing")
+
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(
+            identifier=(glue_table['DatabaseName'], glue_table['Name']),
+            metadata=metadata,
+            metadata_location=metadata_location
+        )
+
+    def _iceberg_to_glue(self, iceberg_table):
+        # TODO
+        pass
+
+    def _construct_parameters(self, metadata_location: str) -> Dict[str, str]:
+        properties = {"table_type": "ICEBERG", "metadata_location": metadata_location}
+        return properties
+
+    def _default_warehouse_location(self, database_name: str, table_name: str):
+        try:
+            response = self.client.get_database(Name=database_name)
+        # TODO: handle response and errors
+        except:
+            raise NoSuchNamespaceError("Database not found")
+
+        if "LocationUri" in response["Database"]:
+            return f"{response['Database']['LocationUri']}/table_name"
+
+        # TODO: should extract warehouse path from the properties and handle potential errors
+        return f"{self.properties['warehouse_path']}/{database_name}.db/{table_name}"
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str):
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    def _write_metadata(self, metadata: TableMetadata, io: FileIO, metadate_path: str):
+        ToOutputFile.table_metadata(metadata, io.new_output(metadate_path))
+
+    # tested on pre-existing database
+    def create_table(
+            self,
+            identifier: Union[str, Identifier],
+            schema: Schema,
+            location: Optional[str] = None,
+            partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+            sort_order: SortOrder = UNSORTED_SORT_ORDER,
+            properties: Properties = EMPTY_DICT,
+    ) -> Table:
```

Comment:
please add comments to the method like https://github.com/apache/iceberg/blob/master/python/pyiceberg/catalog/hive.py#L285

Reply:
similar comment for load_table

Reply:
Currently, I just copy the comment from `hive.py` since the comment just describes the general usage of `Catalog.create_table` and `Catalog.create_table`. Do you think there is some additional information specific to `GlueCatalog` that should be added to the comment?

Reply:
yes this is fine. We can be more specific in the general comment, like `Create an Iceberg table in Glue catalog`